src/toast/ops/detrend.py

```
@@ -0,0 +1,172 @@
+# Copyright (c) 2025-2025 by the parties listed in the AUTHORS file.
+# All rights reserved.  Use of this source code is governed by
+# a BSD-style license that can be found in the LICENSE file.
+import os
+
+import numpy as np
+import traitlets
+
+from ..observation import default_values as defaults
+from ..timing import function_timer
+from ..traits import Int, Unicode, trait_docs
+from ..utils import Logger
+from .operator import Operator
+
+@trait_docs
+class Detrend(Operator):
+    """Remove mean/median/slope of detector data.
+
+    method='mean'   will subtract mean of each detector data.
+
+    method='median' will subtract median of each detector data.
+
+    method='linear' will subtract mean and slope of each detector data,
+    matching `match_nsample` samples at the beginning and end of detector data.
+    """
+
+    # Class traits
+
+    API = Int(0, help="Internal interface version for this operator")
+
+    det_data = Unicode(
+        defaults.det_data,
+        help="Observation detdata key apply detrend to",
+    )
+
+    det_mask = Int(
+        defaults.det_mask_invalid,
+        help="Bit mask value for per-detector flagging",
+    )
+
+    shared_flags = Unicode(
+        defaults.shared_flags,
+        allow_none=True,
+        help="Observation shared key for telescope flags to use",
+    )
+
+    shared_flag_mask = Int(
+        defaults.shared_mask_invalid,
+        help="Bit mask value for optional shared flagging",
+    )
+
+    det_flags = Unicode(
+        defaults.det_flags,
+        allow_none=True,
+        help="Observation detdata key for flags to use",
+    )
+
+    det_flag_mask = Int(
+        defaults.det_mask_invalid,
```

This is certainly an application where we would want to disregard planets, point sources and bright Galactic emission. The default flag mask should be `defaults.det_mask_nonscience`: https://github.com/simonsobs/LAT-scan-schedules/blob/main/iso/phase3/2025-10-22T21%3A36%3A11%2B00%3A00_phase3_type3_lat_field_schedule.txt